JavaScripts/bacteriaDNA.js

```
@@ -0,0 +1,185 @@
+            this.dna.forEach((base, index) => {
+                if (base === otherBacteria.dna[index]) {
+                    count++;
+                }
+            });
+            let percentageOverlap = ((count / this.dna.length) * 100).toFixed(2);
+            console.log(`specimen #${this.specimenNum} and specimen #${otherBacteria.specimenNum
+                } have ${percentageOverlap}% DNA in common.`);
+            return percentageOverlap;
+        },
+        // TASK 4: if either C or G accounts for 60% of the DNA will return true
+        willLikelySurvive() {
+            // counts all C and G bases and returns true if either is 60%
+            let countC = 0;
+            let countG = 0;
+            this.dna.forEach((base) => {
+                if (base === "C") {
+                    countC++;
+                } else if (base === "G") {
+                    countG++;
+                }
+            });
+            // console.log(`C: ${countC}, G: ${countG}`);
+            let percentageC = (countC / this.dna.length) * 100;
+            let percentageG = (countG / this.dna.length) * 100;
+            // console.log(`%C: ${percentageC}, %G: ${percentageG}`)
+            if (percentageC >= 60 || percentageG >= 60) {
+                return true;
+            } else {
+                return false;
+            }
+        },
+        // TASK 6: method that creates complementary DNA strands
+        complementStrand() {
+            return this.dna.map(
+                (base) => {
+                    switch (base) {
+                        case "A": return "T"
+                            break;
+                        case "T": return "A"
+                            break;
+                        case "G": return "C"
+                            break;
+                        case "C": return "G"
+                            break;
+                        default:
+                            break;
```

I'd throw an error on default as it should never happen

default: throw new Error('Invalid base found: ' + base);